istanbul/client.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 AMIS Technologies
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package istanbul
+
+import (
+	"context"
+	"math/big"
+	"sort"
+	"strings"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	ethClient "github.com/getamis/eth-client/client"
+)
+
+type client struct {
+	ethClient.Client
+	c *rpc.Client
+}
+
+func Dial(rawurl string) (Client, error) {
+	rc, err := rpc.Dial(rawurl)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &client{
+		Client: ethClient.NewClient(rc),
+		c:      rc,
+	}
+
+	return c, nil
+}
```

Comment:
Above struct and Dial function are duplicate as default eth client. Not sure if there is a way we can use the same code.

Reply:
No, this is a wrapper based on our client interface
1. this struct should support our Client interface(https://github.com/getamis/eth-client/blob/develop/client/api.go#L29). So we should use `ethClient.NewClient` to get instance. 
2. `quorum` / `istanbul` need to handle their own `rpc.Client`, because the rpc.Client is not exposed in our eth client(i don't want to expose). 

if have any suggestion, please leave a comment for this

Reply:
Got it.